### PR TITLE
Convert markdown processing to a package

### PR DESCRIPTION
**Pull request by jonatanschroeder**
This is part of the process to eventually allow elements to use the same markdown processing used in the server side, using remark/rehype instead of showdown.

The question processing of data (with `pl-code` for code blocks and the `<markdown>` tags) is left in the library. The package is still used to handle the creation of the process.

This does not actually create any new functionality yet. The code is expected to work mostly the same as before. Dependencies were moved to the package where possible.

### CODE REVIEW

File: packages/markdown/package.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  "private": true,
+  "type": "module",
+  "main": "./dist/index.js",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/PrairieLearn/PrairieLearn.git",
+    "directory": "packages/markdown"
+  },
+  "scripts": {
+    "build": "tsc",
```
Comment (jonatanschroeder) [inline packages/markdown/package.json:13]: @nwalters512 it doesn't need to be in this PR, but I'd imagine there would need to be a different build (esbuild? compiled-assets?) for this to work in client side, right?

Comment (nwalters512) [reply packages/markdown/package.json:13]: That's correct; we'd be bundling, not just transpiling.

File: apps/prairielearn/src/lib/markdown.test.ts [hunk 2]
```diff
@@ -2,162 +2,140 @@ import { assert } from 'chai';
 
 import * as markdown from './markdown.js';
 
-const testMarkdownQuestion = (question, expected) => {
+async function testMarkdownQuestion(question, expected) {
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/markdown.test.ts:5]: Why was this changed to an async function?

Consider adding types while you're here.

File: packages/markdown/package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "dev": "tsc --watch --preserveWatchOutput",
+    "test": "c8 mocha src/**/*.test.ts"
+  },
+  "devDependencies": {
```
Comment (nwalters512) [inline packages/markdown/package.json:17]: Conventionally, the order is `dependencies`, then `devDependencies`, then `c8`. If this order is copied from another package, lmk and we can try to fix that one up for consistency.

File: packages/markdown/src/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+async function testMarkdown(
+  original: string,
+  expected: string,
+  options: { inline?: boolean } = {},
+) {
+  const actual = await markdown.markdownToHtml(original, options);
+  assert.equal(actual.toString().trim(), expected);
+}
+
+describe('Markdown processing', () => {
```
Comment (nwalters512) [inline packages/markdown/src/index.test.ts:14]: There seems to be a lot of overlap, if not complete duplication, with the tests in `apps/prairielearn/src/lib/markdown.test.ts`. Can we deduplicate?

File: apps/prairielearn/src/lib/markdown.test.ts (hunk 2 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/markdown.test.ts]: `testMarkdown` was async originally, as I converted it to a regular function (instead of const) I added async, and ended up adding async here too. Changed in f82e6cf49.

File: packages/markdown/src/index.test.ts (hunk 4 above)
Comment (jonatanschroeder) [reply packages/markdown/src/index.test.ts:14]: Deduplicated in f82e6cf49.

File: packages/markdown/package.json (hunk 3 above)
Comment (jonatanschroeder) [reply packages/markdown/package.json:29]: This was copied from path-utils, which doesn't have `dependencies`, so the block was added by `yarn add` at the end. Reordered in e462cf637.

**Comment by nwalters512** [comment]
@jonatanschroeder there are merge conflicts that need to be resolved here.

**Comment by jonatanschroeder** [comment]
> @jonatanschroeder there are merge conflicts that need to be resolved here.

Handled. I also updated the package dependencies to match those recently updated in the app.

File: yarn.lock [hunk 5]
```diff
@@ -6339,6 +6360,27 @@ __metadata:
   languageName: node
   linkType: hard
 
+"c8@npm:^9.1.0":
```
Comment (nwalters512) [inline yarn.lock:6363]: Could you bump this to 10.x so that we can us the same version that's already used everywhere else?

Comment (jonatanschroeder) [reply yarn.lock]: Done.

**Review by nwalters512 — APPROVED** [review]
